docs/usage/annotations.rst

```diff
@@ -0,0 +1,118 @@
+
+.. warning::
+    The `whole line <https://openqasm.com/language/directives.html#annotations#:~:text=continue%20to%20the%20end%20of%20the%20line>`_ will be interpreted like an annotation.
+    Therefore you cannot use inline-comments on annotations!
+
+Input
+-----
+
+One input is defined as a single qubit register (:class:`~openqasm3.ast.QubitDeclaration`) with a single `@leqo.input` annotation.
+
+* Inputs can be of arbitrary size (See :ref:`input-memory-layout`)
+* Inputs have to be defined as contiguous memory
+    One :class:`~openqasm3.ast.QubitDeclaration` corresponds to one input
+* Input indices must be selected from a contiguous range of integers starting at `0`
+   No skips, no duplicates
+* Inputs may be declared anywhere in code
+* Input annotations may only appear on a :class:`~openqasm3.ast.QubitDeclaration`
+* Input annotations may only appear once per statement
+
+.. code-block:: openqasm3
+    :linenos:
+
+    // Qubit array
+    @leqo.input <<InputIndex>>
+    qubit[<<length>>] someName;
+
+    // Single qubit
+    @leqo.input <<InputIndex>>
+    qubit someName;
+
+.. _input-memory-layout:
+
+Memory Layout
+~~~~~~~~~~~~~
+
+There is no explicit limit to the size (qubit count) of an input.
+Keep in mind though that the number of physical qubits on a real device is limited.
+See :ref:`reusable-qubit-annotation` for workarounds.
+
+The annotated input size must match the input size of the corresponding node from the `visual model <https://github.com/LEQO-Framework/low-code-modeler>`_.
+
+The backend actively ensures that input memory is initialized.
+All other qubits still have to be assumed to be in an undefined state (See `OpenQasm Specification <https://openqasm.com/language/types.html#qubits#:~:text=Qubits%20are%20initially%20in%20an%20undefined%20state>`_ and :ref:`reusable-qubit-annotation`).
+
+In the future, it will be allowed to input less qubits than specified using the annotation.
+In this case the backend will fill the lowest bytes with the actual input and ensure the upper bytes are initialized to zero:
+
+    .. csv-table:: Example input register of size `7`
+        :header: "0", "1", "2", "3", "4", "5", "6"
+
+        "p[0]", "p[1]", "p[2]", "p[3]", "p[4]", "p[5]", "p[6]"
+        "p[0]", "p[1]", "\|0⟩", "\|0⟩", "\|0⟩", "\|0⟩", "\|0⟩"
+
+.. note::
+    The input might be split into separate non-contiguous memory blocks by the backend.
+
+Output
+------
+
+One output is defines as a single alias (:class:`~openqasm3.ast.AliasStatement`) with a single `@leqo.output` annotation.
+
+* Outputs may be concatenated from multiple non-contiguous blocks of memory.
+* Output indices must be selected from a contiguous range of integers starting at `0`
+    No skips, no duplicates
+* Outputs may be declared anywhere in code
+* Outputs may be used like any other alias
+* Output annotations may only appear above a :class:`~openqasm3.ast.AliasStatement` pointing to qubits
+* Output annotations may only appear once per statement
```

Comment:
And Qubits may only be used/aliased in one or no outputs (not in multiple outputs)